subprojects/groovy-groovysh/src/main/groovy/org/codehaus/groovy/tools/shell/completion/ImportsSyntaxCompletor.groovy

```diff
@@ -87,51 +78,49 @@ class ImportsSyntaxCompletor implements IdentifierCompletor {
         return foundMatch
     }
 
-    private static final String STATIC_IMPORT_PATTERN = ~/^import static ([a-z0-9]+\.)+[A-Z][a-zA-Z0-9]*(\.(\*|[^.]+))?$/
+    private static final String STATIC_IMPORT_PATTERN = ~/^static ([a-zA-Z_][a-zA-Z_0-9]*\.)+([a-zA-Z_][a-zA-Z_0-9]*|\*)$/
 
     /**
      * finds matching imported classes or static methods
-     * @param prefix
-     * @param importSpec
-     * @param matches
-     * @return
+     * @param importSpec an import statement without the leading 'import ' or trailing semicolon
+     * @return all names matching the importSpec
      */
-    void collectImportedSymbols(final String importSpec, final Collection<String> matches) {
+    SortedSet<String> collectImportedSymbols(final String importSpec) {
```

Comment:
we decided to be strict about such things in a bugfix release and 2.4.8 is one. So yeah, unless the method signature is made compatible no 2.4.x